x-pack/plugins/security_solution/server/lib/detection_engine/rule_types/query/create_query_alert_type.ts

```diff
@@ -97,6 +101,16 @@ export const createQueryAlertType = (
         primaryTimestamp,
         secondaryTimestamp,
       });
+
+      if (responseActions?.length && result.createdSignalsCount) {
+        scheduleNotificationResponseActions(
+          {
+            signals: result.createdSignals,
+            responseActions: responseActions as unknown as ResponseAction[],
+          },
+          osqueryCreateAction
+        );
+      }
```

Comment:
https://github.com/elastic/kibana/pull/140064 is going to make the `query` and `saved_query` rule types have the same implementation at the framework level so users can switch back and forth between the 2 types. The result is that a `query` rule from the users perspective could be using the `siem.savedQueryRule` rule type behind the scenes.

I would suggest moving this block into the `queryExecutor` function, as that's shared between the 2 rule types and will ensure the response actions are executed for all `query` rule types.

Reply:
Ooo, good hint, thank you, will move the code :)